skeleton/src/main/java/com/dropbox/kaiken/skeleton/skeleton/usermanagement/UserManager.kt

```diff
@@ -0,0 +1,119 @@
+package com.dropbox.kaiken.skeleton.skeleton.usermanagement
+
+import com.dropbox.kaiken.skeleton.skeleton.usermanagement.auth.ActiveUserManager
+import com.dropbox.kaiken.skeleton.skeleton.usermanagement.auth.SkeletonAuthInteractor
+import com.dropbox.kaiken.scoping.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.drop
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.scan
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import javax.inject.Inject
+
+/**
+ * Manages the set of authenticated users for an app.
+ */
+interface UserManager {
+    /**
+     * Sets the given user ID to the active user. This returns true if the user ID is known, false otherwise.
+     */
+    suspend fun setActiveUser(newActiveUserId: String): Boolean
+
+    /**
+     * Returns the active user. If there is no active user, this will return null.
+     */
+    suspend fun getActiveUser(): User?
+
+    /**
+     * Provides a flow of the [UserState].  Observe this for changes related to users added, removed, and active users updated.
+     *
+     */
+    fun getUserState(): Flow<UserState>
```

Comment:
I know that you're going to yell at me. But I'd highly recommend here using a traditional listener architecture. For example, having a `UserManagerListener` that has an `onCurrentUserChanged`. And adding and removing listeners to the `UserManager`.

This `UserManager` code is easy to get wrong, so you need to make it very easy to debug. But you do you :)

Reply:
Ha I actually made similar argument to @chris-mitchell keep it simple or make it a store which I know won't go wrong. Right now its unclear if it is events or state, similar what the backing interactor is returning. He's thinking of how to approach and will take this up in follow up diff.

Reply:
I think the big piece here is that this isn't the actual `UserManager` as it won't ultimately persist the user information to disk.  We definitely need to improve the API on this though to be a generic solution.  This was mostly just moved from another impl.